code/game/gamemodes/cult/runes.dm

```diff
@@ -1,1044 +1,944 @@
+structure_check() searches for nearby cultist structures required for the invocation. Proper structures are pylons, forges, archives, and altars.
+*/
 
-/obj/effect/rune/proc/tomesummon()
-	if(istype(src,/obj/effect/rune))
-		usr.say("N[pick("'","`")]ath reth sh'yro eth d'raggathnor!")
-	else
-		usr.whisper("N[pick("'","`")]ath reth sh'yro eth d'raggathnor!")
-	usr.visible_message("\red Rune disappears with a flash of red light, and in its place now a book lies.", \
-	"\red You are blinded by the flash of red light! After you're able to see again, you see that now instead of the rune there's a book.", \
-	"\red You hear a pop and smell ozone.")
-	if(istype(src,/obj/effect/rune))
-		new /obj/item/weapon/tome(src.loc)
-	else
-		new /obj/item/weapon/tome(usr.loc)
+/obj/effect/rune/proc/can_invoke(var/mob/living/user)
+	//This proc determines if the rune can be invoked at the time. If there are multiple required cultists, it will find all nearby cultists.
+	var/list/invokers = list() //people eligible to invoke the rune
+	var/list/chanters = list() //people who will actually chant the rune when passed to invoke()
+	chanters |= user
+	invokers |= user
+	if(req_cultists > 1 || allow_excess_invokers)
+		for(var/mob/living/L in range(1, src))
+			if(iscultist(L))
+				if(L == user)
+					continue
+				if(ishuman(L))
+					var/mob/living/carbon/human/H = L
+					if((H.disabilities & MUTE) || H.silent)
+						continue
+				if(L.stat)
+					continue
+				invokers |= L
+		if(invokers.len >= req_cultists)
+			if(allow_excess_invokers)
+				chanters |= invokers
+			else
+				invokers -= user
+				shuffle(invokers)
+				for(var/i in 0 to req_cultists)
+					var/L = pick_n_take(invokers)
+					chanters |= L
+	return chanters
+
+/obj/effect/rune/proc/invoke(var/list/invokers)
+	//This proc contains the effects of the rune as well as things that happen afterwards. If you want it to spawn an object and then delete itself, have both here.
+	if(invocation)
+		for(var/M in invokers)
+			var/mob/living/L = M
+			if(!L.IsVocal())//MIMES
+			//do emote here...
+				L.emote("acts out the rune invocation.")//it will looks like flailing...
+			else
+				L.say(invocation)
+	var/oldtransform = transform
+	spawn(0) //animate is a delay, we want to avoid being delayed
+		animate(src, transform = matrix()*2, alpha = 0, time = 5) //fade out
+		animate(transform = oldtransform, alpha = 255, time = 0)
+
+/obj/effect/rune/proc/fail_invoke(var/mob/living/user)
+	//This proc contains the effects of a rune if it is not invoked correctly, through either invalid wording or not enough cultists. By default, it's just a basic fizzle.
+	visible_message("<span class='warning'>The markings pulse with a small flash of red light, then fall dark.</span>")
+	spawn(0) //animate is a delay, we want to avoid being delayed
+		animate(src, color = rgb(255, 0, 0), time = 0)
```

Comment:
this animate 
it doesn't have a point with the next animate including the src parameter